nix/default.nix

```diff
@@ -0,0 +1,48 @@
+{ system ? builtins.currentSystem
+, crossSystem ? null
+, config ? {}
+, sourcesOverride ? {}
+}:
+let
+  sources = import ./sources.nix { inherit pkgs; }
+    // sourcesOverride;
+  iohKNix = import sources.iohk-nix {};
+  haskellNix = import sources."haskell.nix";
+  # use our own nixpkgs if it exists in our sources,
+  # otherwise use iohkNix default nixpkgs.
+  nixpkgs = if (sources ? nixpkgs)
+    then (builtins.trace "Not using IOHK default nixpkgs (use 'niv drop nixpkgs' to use default for better sharing)"
+      sources.nixpkgs)
+    else (builtins.trace "Using IOHK default nixpkgs"
+      iohKNix.nixpkgs);
+
+  # for inclusion in pkgs:
+  overlays =
+    # Haskell.nix (https://github.com/input-output-hk/haskell.nix)
+    haskellNix.overlays
+    # haskell-nix.haskellLib.extra: some useful extra utility functions for haskell.nix
+    ++ iohKNix.overlays.haskell-nix-extra
+    # iohkNix: nix utilities and niv:
+    ++ iohKNix.overlays.iohkNix
+    # our own overlays:
+    ++ [
+      (pkgs: _: with pkgs; {
+
+        # commonLib: mix pkgs.lib with iohk-nix utils and our own:
+        commonLib = lib // iohkNix
+          // import ./util.nix { inherit lib haskell-nix; }
+          # also expose our sources and overlays
+          // { inherit overlays sources; };
+      })
+      # And, of course, our haskell-nix-ified cabal project:
+      (import ./pkgs.nix)
```

Comment:
Perhaps this is a controversial opinion, but I don't actually *want* our Plutus packages injected into `nixpkgs`. Why would we do that?

Reply:
Ahh c'mon, some global `state`, what could possibly go wrong? ;-) would you prefer to pass it everywhere?
One of the upshots of having it in here is that you can apply additional overlays after it, if you needed to, and the interface for consumers would be identical. (They'd just pick the plutusHaskellPackages from pkgs).